Apps/phredmine/redmine_connector.py

```diff
@@ -0,0 +1,996 @@
+
+        # Action Result
+        action_result.add_data(response)
+
+        return action_result.set_status(phantom.APP_SUCCESS, REDMINE_SUCC_UPDATE_TICKET)
+
+    def _handle_add_comment(self, param):
+        """Creates a comment on an existing ticket"""
+        self.save_progress(
+            "In action handler for: {0}".format(self.get_action_identifier())
+        )
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        # Parameters
+        id = param["id"]
+        comment = param["comment"]
+
+        payload = {"issue": {"notes": comment}}
+
+        ret_val, response = self._make_rest_call(
+            f"/issues/{id}.json",
+            action_result,
+            method="put",
+            params=None,
+            headers=None,
+            json=payload,
+        )
+
+        if phantom.is_fail(ret_val):
+            return action_result.set_status(phantom.APP_ERROR, REDMINE_ERR_ADD_COMMENT.format(id=id))
+
+        ret_val, response = self._retrieve_ticket(action_result, id)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        # Action Result
+        action_result.add_data(response)
+
+        return action_result.set_status(phantom.APP_SUCCESS, REDMINE_SUCC_ADD_COMMENT)
+
+    def _handle_list_tickets(self, param):
+        """Retrieves a list of tickets"""
+        self.save_progress(
+            "In action handler for: {0}".format(self.get_action_identifier())
+        )
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        # Parameters
+        query = param.get("query", "")
+
+        # Integer validation for 'start_index' action parameter
+        start_index = param.get("start_index", 0)
+        ret_val, start_index = self._validate_integer(action_result, start_index, REDMINE_START_INDEX_KEY, allow_zero=True)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        # Integer validation for 'max_results' action parameter
+        max_results = param.get("max_results", 100)
+        ret_val, max_results = self._validate_integer(action_result, max_results, REDMINE_MAX_RESULTS_KEY, allow_zero=True)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        params = {
+            "offset": start_index,
+            "limit": max_results,
+            "project_id": self._project_id
+        }
+
+        ret_val, response = self._make_rest_call(
+            f"/issues.json{query}", action_result, params=params, headers=None
+        )
+
+        if phantom.is_fail(ret_val) or "issues" not in response:
+            return action_result.set_status(
+                phantom.APP_ERROR,
+                REDMINE_ERR_LIST_TICKETS
+            )
+
+        # Action Result
+        action_result.add_data(response)
+
+        summary = action_result.update_summary({})
+        summary['num_tickets'] = len(response['issues'])
```

Comment:
Please use .get() method for API response.

Reply:
Not required for the 'issues' key as we have kept a check for the key before adding it in summary.
`if phantom.is_fail(ret_val) or "issues" not in response:`
However, I have handled similar instances in the code.